src/core/util.cpp

```diff
@@ -754,31 +754,47 @@ namespace mamba
     LockFile::LockFile(const fs::path& path)
         : m_path(path)
     {
-        LOG_INFO << "Locking " << path;
+        if (!fs::exists(path))
+        {
+            LOG_DEBUG << "Could not lock non existing path '" << path.string() << "'";
+            m_fd = -1;
+            return;
+        }
 
-        m_fd = open(path.c_str(), O_RDWR | O_CREAT, 0666);
+        LOG_DEBUG << "Locking '" << path.string() << "'";
+        if (fs::is_directory(path))
+            m_lock = m_path / "mamba.lock";
+        else
+            m_lock = m_path;
+
+        m_fd = open(m_lock.c_str(), O_RDWR | O_CREAT, 0666);
 
         if (m_fd <= 0)
         {
-            LOG_ERROR << "Could not open lockfile " << path;
+            LOG_DEBUG << "Could not open lockfile '" << m_lock.string() << "'";
         }
         else
         {
+            LOG_TRACE << "Lockfile created at '" << m_lock.string() << "'";
             m_pid = getpid();
             int ret = try_lock(m_fd, m_pid, false);
             if (ret != m_pid)
             {
-                LOG_WARNING << "Cannot lock file " << path
-                            << "\nWaiting for other mamba process to finish.\n";
+                LOG_WARNING << "Cannot lock '" << m_path << "'"
```

Comment:
usually the quoting is done automatically by `fs::path`.

Reply:
I just try to have homogeneous logs, using the same syntax everywhere

Reply:
if `m_path` is a `fs::path` then the `operator<<` calls `std::quoted`. So you don't need to add `'`